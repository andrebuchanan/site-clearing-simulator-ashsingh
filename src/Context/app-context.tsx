import React, { ReactNode, useMemo, useState } from 'react'

type Session = {
  init: boolean
  quit: boolean
}

interface AppContextInterface {
  fuelConsumed: number[]
  itemizedCosts: number[]
  loggedCommands: string[]
  sessionState: Session
  siteData: string[][]
  logCommand: (command: string) => void
  saveSiteData: (data: string[][]) => void
  updateCost: (price: number) => void
  updateFuel: (quantity: number) => void
  updateSession: (session: Session) => void
}

export const AppContext = React.createContext<AppContextInterface>({
  fuelConsumed: [],
  itemizedCosts: [],
  loggedCommands: [],
  sessionState: { init: true, quit: false },
  siteData: [],
  logCommand: () => undefined,
  saveSiteData: () => undefined,
  updateCost: () => undefined,
  updateFuel: () => undefined,
  updateSession: () => undefined
})

export const AppContextProvider = ({ children }: { children: ReactNode }) => {
  const [commands, setCommands] = useState<string[]>([])
  const [cost, setCost] = useState<number[]>([])
  const [fuel, setFuel] = useState<number[]>([])
  const [session, setSession] = useState<Session>({ init: true, quit: false })
  const [siteInput, setSiteInput] = useState<string[][]>([])

  const logCommandHandler = (command: string) => {
    setCommands((prevState) => [...prevState, command])
  }

  const saveSiteDataHandler = (data: string[][]) => {
    setSiteInput(data)
  }

  const updateCostHandler = (price: number) => {
    setCost((prevState) => [...prevState, price])
  }

  const updateFuelHandler = (quantity: number) => {
    setFuel((prevState) => [...prevState, quantity])
  }

  const updateSessionHandler = (sessionProp: Session) => {
    console.log('session', sessionProp)
    if (sessionProp.quit) console.log('quit')
    setSession(sessionProp)
  }

  const contextValue = useMemo(
    () => ({
      fuelConsumed: fuel,
      itemizedCosts: cost,
      loggedCommands: commands,
      sessionState: session,
      siteData: siteInput,
      logCommand: logCommandHandler,
      saveSiteData: saveSiteDataHandler,
      updateCost: updateCostHandler,
      updateFuel: updateFuelHandler,
      updateSession: updateSessionHandler
    }),
    [commands, cost, fuel, session, siteInput]
  )

  return (
    <AppContext.Provider value={contextValue}>{children}</AppContext.Provider>
  )
}

export default AppContextProvider
