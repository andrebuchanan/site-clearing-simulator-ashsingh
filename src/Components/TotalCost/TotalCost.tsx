import React, { useContext } from 'react'

import Typography from '@mui/material/Typography'

import { AppContext } from '../../Context'

export const TotalCost = () => {
  const { itemizedCosts } = useContext(AppContext)
  const totalCost = itemizedCosts.reduce((acc, i) => acc + i, 0)
  return (
    <Typography variant="h6" component="div" gutterBottom>
      Total Cost: {totalCost} Credits
    </Typography>
  )
}

export default TotalCost
