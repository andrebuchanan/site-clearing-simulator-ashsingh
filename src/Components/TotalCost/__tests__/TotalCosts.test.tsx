import React from 'react'
import { render, screen } from '@testing-library/react'

import { AppContext } from '../../../Context'
import TotalCost from '../TotalCost'

const contextValue = {
  itemizedCosts: [1, 5, 3]
}

test('renders TotalCost', () => {
  const { asFragment } = render(
    <AppContext.Provider value={contextValue}>
      <TotalCost />
    </AppContext.Provider>
  )

  expect(asFragment()).toMatchSnapshot()
  expect(screen.getByText(/9 Credits/i)).toBeDefined()
})
