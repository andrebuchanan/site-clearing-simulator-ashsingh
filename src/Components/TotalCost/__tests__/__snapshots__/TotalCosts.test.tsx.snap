// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders TotalCost 1`] = `
<DocumentFragment>
  <div
    class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
  >
    Total Cost: 9 Credits
  </div>
</DocumentFragment>
`;
